File: epi_judge_python/fibonacci.py
from test_framework import generic_test


def fibonacci(n):
    f = 0
    s = 1

    if n <= 1:
        return n
    else:
        for i in range(2,n + 1):
            tmp = f
            f = s
            s += tmp

    return s


if __name__ == '__main__':
    exit(
        generic_test.generic_test_main("fibonacci.py", 'fibonacci.tsv',
                                       fibonacci))

File: epi_judge_java/epi/Fibonacci.java
package epi;
import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;
public class Fibonacci {

  @EpiTest(testDataFile = "fibonacci.tsv")

  public static int fibonacci(int n) {
    if (n <= 1){
      return n;
    }
    int a = 0, b = 1, i = 2;

    while (i <= n){
      int tmp = b;
      b = a+b;
      a = tmp;
      i ++;
    }
    return b;
  }

  public static void main(String[] args) {
    System.exit(
        GenericTest
            .runFromAnnotations(args, "Fibonacci.java",
                                new Object() {}.getClass().getEnclosingClass())
            .ordinal());
  }
}
